### PR TITLE
Add org.tug.texworks

### PR DESCRIPTION
**Pull request by stloeffler**

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.tug.texworks

File: org.tug.texworks.yaml [hunk 1]
```diff
@@ -0,0 +1,143 @@
+app-id: org.tug.texworks
+runtime: org.kde.Platform
+runtime-version: '5.12'
```
Comment (LeandroStanger) [inline org.tug.texworks.yaml]: Update runtime to 5.13

File: org.tug.texworks.yaml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+sdk: org.kde.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.texlive
+command: texworks
+rename-desktop-file: texworks.desktop
+rename-appdata-file: texworks.appdata.xml
```
Comment (LeandroStanger) [inline org.tug.texworks.yaml]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Comment (bilelmoussaoui) [reply org.tug.texworks.yaml]: No, this doesn't work for rename-appdata-file.

File: org.tug.texworks.yaml (hunk 1 above)
Comment (stloeffler) [reply org.tug.texworks.yaml]: Done in c4d3335

**Comment by LeandroStanger** [comment]
bot, build org.tug.texworks

**Comment by stloeffler** [comment]
From a quick first test, the program seems to work as expected.

File: org.tug.texworks.yaml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/TeXworks/manual/releases/download/2019-03-16/TeXworks-manual-html.zip
+        sha256: 34ed73d0996dd15bf8c2844712ac882be2901257dfb6f31df0b5adb0f2c4ff18
+    buildsystem: simple
+    build-commands:
+      - install -d /app/share/doc/texworks-help
+      - cp -ra * /app/share/doc/texworks-help
+
+  - name: icons # as per flathub's app requirements
```
Comment (barthalion) [inline org.tug.texworks.yaml]: Please move it to `texworks` module, there's no need to make it separate one.

File: org.tug.texworks.yaml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    sources:
+      - type: file
+        path: TeXworks-64.png
+      - type: file
+        path: TeXworks-128.png
+    buildsystem: simple
+    build-commands:
+      - install -D TeXworks-64.png /app/share/icons/hicolor/64x64/apps/TeXworks.png
```
Comment (barthalion) [inline org.tug.texworks.yaml]: Are these files not available in upstream repository?

**Comment by barthalion** [comment]
Just some nitpicks. Should we invite someone besides you to collaborators of new repo?

File: org.tug.texworks.yaml [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+modules:
+  - name: texlive
+    buildsystem: simple
+    build-commands:
+      # Don't use /usr/lib/sdk/texlive/install.sh as that copies everthing into
+      # /app rather than a subdirectory, causing massive clutter of that folder
+      - install -d /app/texlive
+      - cp -ra /usr/lib/sdk/texlive/* /app/texlive
+
+  - name: openjpeg # required by poppler
```
Comment (jakobjakobson13) [inline org.tug.texworks.yaml]: Could you use the openjpg module from the shared modules here?

File: org.tug.texworks.yaml (hunk 4 above)
Comment (stloeffler) [reply org.tug.texworks.yaml]: They were added to upstream master quite recently, but unfortunately are not available in the lastest stable version yet.

File: org.tug.texworks.yaml (hunk 3 above)
Comment (stloeffler) [reply org.tug.texworks.yaml]: Done in 978ddea

Comment (stloeffler) [reply org.tug.texworks.yaml]: Done in 978ddea

File: org.tug.texworks.yaml (hunk 5 above)
Comment (stloeffler) [reply org.tug.texworks.yaml]: Done in 952cb5e

**Comment by stloeffler** [comment]
There's probably no need ATM to invited any collaborators to the new repo as I am currently the primary maintainer of the program.

**Comment by bilelmoussaoui** [comment]
bot, build org.tug.texworks

**Comment by bilelmoussaoui** [comment]
You can check https://github.com/flathub/flathub/wiki/App-Maintenance for often asked questions about maintaining an app on Flathub

**Pull request closed without merging**